### PR TITLE
Add ESIL for few ARM instructions and fix minor issues

### PR DESCRIPTION
**Pull request by mrmacete**
Adds:
smmla, smmlar, umull, mls, mla, mvn, bfi, bfc

Fixes:
- ldm index displace
- crop 32 bit math results to 32 bits
- crop 32 bit addresses to 32 bits

related tests here: https://github.com/radare/radare2-regressions/pull/943

### CODE REVIEW

File: libr/anal/p/anal_arm_cs.c [hunk 1]
```diff
@@ -1452,7 +1447,7 @@ r4,r5,r6,3,sp,[*],12,sp,+=
 		const char *comma = "";
 		for (i=1; i<insn->detail->arm.op_count; i++) {
 			r_strbuf_appendf (&op->esil, "%s%s,%d,+,[4],%s,=",
-				comma, ARG (0), i*4, REG (i));
+				comma, ARG (0), (i-1)*4, REG (i));
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: missing spaces around * and -

File: libr/anal/p/anal_arm_cs.c [hunk 2]
```diff
@@ -1313,15 +1313,10 @@ static void arm32math(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 	if (negate) {
 		r_strbuf_appendf (&op->esil, ",-1,^");
 	}
-	// left operand
-	if (strcmp(dest, op1) == 0) {
-		r_strbuf_appendf (&op->esil, ",%s,%s=", dest, opchar);
+	if (strcmp(op1, "pc") == 0) {
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: use !strcmp and add missing space before (

**Comment by radare** [comment]
lgtm, maybe add some tests after fixing the indentation issues? thanks!

**Comment by mrmacete** [comment]
tests are here: https://github.com/radare/radare2-regressions/pull/943 what's missing?

**Pull request merged**